providers/google/semantic_location_history_test.go

```diff
@@ -0,0 +1,154 @@
+
+	p := google{Database: provider.NewDatabase(db)}
+	require.NoError(t, p.Migrate())
+
+	require.NoError(t, p.importSemanticLocationHistoryFromDirectory("testdata/google/Location History/Semantic Location History/"))
+
+	var placeVisits []PlaceVisit
+	var activitySegments []ActivitySegment
+	require.NoError(t, db.
+		Preload("OtherCandidateLocations").
+		Preload("SimplifiedRawPathPoints").
+		Preload("ChildVisits.OtherCandidateLocations").
+		Order("place_visit_id asc").
+		Find(&placeVisits).
+		Error)
+	require.NoError(t, db.
+		Preload("Activities").
+		Preload("SimplifiedRawPathPoints").
+		Preload("TransitStops").
+		Preload("Waypoints").
+		Find(&activitySegments).
+		Error)
+
+	require.Len(t, placeVisits, 2)
+	require.Len(t, activitySegments, 1)
+
+	expectedPlaceVisit := PlaceVisit{
+		CenterLatE7:                 340154991,
+		CenterLngE7:                 -1184860798,
+		DurationEndTimestampMs:      types.DateTime(time.Date(2020, 2, 4, 18, 12, 44, 675000000, time.UTC)),
+		DurationStartTimestampMs:    types.DateTime(time.Date(2020, 2, 4, 17, 39, 12, 182000000, time.UTC)),
+		EditConfirmationStatus:      "NOT_CONFIRMED",
+		LocationAddress:             "1351 3rd Street Promenade\nSanta Monica, CA 90401\nUSA",
+		LocationLatitudeE7:          340165480,
+		LocationLocationConfidence:  55.464928,
+		LocationLongitudeE7:         -1184961984,
+		LocationName:                "Downtown Santa Monica",
+		LocationPlaceID:             "ChIJFwxnxM-kwoARewIypbH6UsU",
+		LocationSourceInfoDeviceTag: 59293564,
+		PlaceConfidence:             "MEDIUM_CONFIDENCE",
+		VisitConfidence:             75,
+		SimplifiedRawPathPoints: []PlacePathPoint{
+			{AccuracyMeters: 24, LatE7: 340470606, LngE7: -1185254605, Time: types.DateTime(time.Date(2020, 2, 11, 17, 54, 14, 505000000, time.UTC))},
+		},
+		OtherCandidateLocations: []CandidateLocation{},
+		ChildVisits: []*PlaceVisit{
+			{
+				CenterLatE7:                 340154991,
+				CenterLngE7:                 -1184860798,
+				DurationEndTimestampMs:      types.DateTime(time.Date(2020, 2, 4, 18, 12, 44, 675000000, time.UTC)),
+				DurationStartTimestampMs:    types.DateTime(time.Date(2020, 2, 4, 17, 39, 12, 182000000, time.UTC)),
+				EditConfirmationStatus:      "NOT_CONFIRMED",
+				LocationAddress:             "1654 Lincoln Blvd\nSanta Monica, CA 90404\nUSA",
+				LocationLatitudeE7:          340157520,
+				LocationLocationConfidence:  25.894972,
+				LocationLongitudeE7:         -1184872646,
+				LocationName:                "Box 'N Burn Santa monica Boxing Fitness gym",
+				LocationPlaceID:             "ChIJD0XpyNKkwoARDAiHsXtjcvg",
+				LocationSourceInfoDeviceTag: 59293564,
+				PlaceConfidence:             "LOW_CONFIDENCE",
+				VisitConfidence:             75,
+				OtherCandidateLocations: []CandidateLocation{
+					{LatitudeE7: 340154424, LocationConfidence: 11.792331, LongitudeE7: -1184863484, PlaceID: "ChIJj0JGzdKkwoARw0F60V1_eQE"},
+				},
+			},
+		},
+	}
+	placeVisits[0].Model = gorm.Model{}
```

Comment:
how's about introducing a helper function for resetting the `gorm.Model` field for visit and all its children recursively?